pkg/util/helper/mcs_test.go

```diff
@@ -0,0 +1,212 @@
+			args: args{
+				client: fake.NewClientBuilder().WithScheme(gclient.NewSchema()).WithObjects(
+					&discoveryv1.EndpointSlice{ObjectMeta: metav1.ObjectMeta{Name: "eps", Namespace: "ns"}}).Build(),
+				endpointSlice: &discoveryv1.EndpointSlice{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "eps", Namespace: "ns",
+						Labels: map[string]string{"foo": "foo1"},
+					},
+					AddressType: discoveryv1.AddressTypeIPv4,
+					Endpoints: []discoveryv1.Endpoint{{
+						Addresses: []string{"1.1.1.1"},
+					}},
+					Ports: []discoveryv1.EndpointPort{{
+						Port: pointer.Int32(80),
+					}},
+				},
+			},
+			want: &discoveryv1.EndpointSlice{
+				TypeMeta: metav1.TypeMeta{APIVersion: "discovery.k8s.io/v1", Kind: "EndpointSlice"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "eps", Namespace: "ns",
+					Labels: map[string]string{"foo": "foo1"},
+				},
+				AddressType: discoveryv1.AddressTypeIPv4,
+				Endpoints: []discoveryv1.Endpoint{{
+					Addresses:  []string{"1.1.1.1"},
+					Conditions: discoveryv1.EndpointConditions{},
+				}},
+				Ports: []discoveryv1.EndpointPort{{
+					Port: pointer.Int32(80),
+				}},
+			},
+			wantErr: false,
+		},
+		{
+			name: "exist and not update",
+			args: args{
+				client: fake.NewClientBuilder().WithScheme(gclient.NewSchema()).WithObjects(
+					&discoveryv1.EndpointSlice{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "eps", Namespace: "ns",
+							Labels: map[string]string{"foo": "foo1"},
+						},
+						AddressType: discoveryv1.AddressTypeIPv4,
+						Endpoints: []discoveryv1.Endpoint{{
+							Addresses: []string{"1.1.1.1"},
+						}},
+						Ports: []discoveryv1.EndpointPort{{
+							Port: pointer.Int32(80),
+						}},
+					}).Build(),
+				endpointSlice: &discoveryv1.EndpointSlice{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "eps", Namespace: "ns",
+						Labels: map[string]string{"foo": "foo1"},
+					},
+					AddressType: discoveryv1.AddressTypeIPv4,
+					Endpoints: []discoveryv1.Endpoint{{
+						Addresses: []string{"1.1.1.1"},
+					}},
+					Ports: []discoveryv1.EndpointPort{{
+						Port: pointer.Int32(80),
+					}},
+				},
+			},
+			want: &discoveryv1.EndpointSlice{
+				TypeMeta: metav1.TypeMeta{APIVersion: "discovery.k8s.io/v1", Kind: "EndpointSlice"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "eps", Namespace: "ns",
+					Labels: map[string]string{"foo": "foo1"},
+				},
+				AddressType: discoveryv1.AddressTypeIPv4,
+				Endpoints: []discoveryv1.Endpoint{{
+					Addresses:  []string{"1.1.1.1"},
+					Conditions: discoveryv1.EndpointConditions{},
+				}},
+				Ports: []discoveryv1.EndpointPort{{
+					Port: pointer.Int32(80),
+				}},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := CreateOrUpdateEndpointSlice(tt.args.client, tt.args.endpointSlice); (err != nil) != tt.wantErr {
+				t.Errorf("CreateOrUpdateEndpointSlice() error = %v, wantErr %v", err, tt.wantErr)
```

Comment:
if  want to stop directly, use `t.Fatalf`

Reply:
Golang generate the test  function with `t.Error`. I don't modify it.

Reply:
I check the usage of Errorf and Fatalf: https://ieftimov.com/posts/testing-in-go-failing-tests/#when-to-use-errorf-or-fatalf